kernel/debug/dbgio.c

```diff
@@ -151,39 +152,34 @@ int dbgio_write_buffer_xlat(const uint8 *data, int len) {
 int dbgio_write_str(const char *str) {
     if(dbgio_enabled) {
         assert(dbgio);
-        return dbgio_write_buffer_xlat((const uint8*)str, strlen(str));
+        return dbgio_write_buffer_xlat((const uint8_t *)str, strlen(str));
     }
 
     return -1;
 }
 
-// Not re-entrant
-static char printf_buf[1024];
-static spinlock_t lock = SPINLOCK_INITIALIZER;
+int dbgio_vprintf(const char *fmt, va_list *var_args) { 
+    char buffer[1024];
+    int i = vsnprintf(buffer, sizeof(buffer), fmt, *var_args);
+
+    if(i >= (int)sizeof(buffer))
```

Comment:
Shouldn't this trigger a growth of the buffer? At least it's better than before where it won't just overflow print_buf. Thinking something like:

    char *overflow = &buffer;
    int i = vsnprintf( all that );
    
    if(i >= sizeof) {
        overflow = malloc(i + 1);
        vsnprintf (overflow, i, fmt, *var_args);
    }
    dbgio_write(overflow)
    
    if(overflow != &buffer)
       free(overflow);
  

And then also should be error checking vsnprintf for possible negative return for encoding issues.